### ISSUE TITLE
I don't think this is correct. (Correct me if I'm wrong)

### ISSUE THREAD

**Issue by nosferatu500** [pmndrs/eslint-plugin-valtio#32]
```
const { initialized, loading } = useSnapshot(EventsStore);      <-- warning

render (
    {(loading || !initialized) && <Loader text={t("Loading...")} />}
)
```

  valtio/state-snapshot-rule
  warning  Better to just use proxy state
  valtio v0.5.1

**Comment by nosferatu500** [comment]
https://github.com/pmndrs/eslint-plugin-valtio#proxies-in-render-phase-is-not-recommended

**Comment by dai-shi** [comment]
Sounds unexpected, but can you elaborate a bit?
Also, was it working okay with v0.4.x?

**Comment by nosferatu500** [comment]
Yes, this works fine with v0.4.4, but seems to be a false positive with v0.5.1.

**Comment by dai-shi** [comment]
Let's wait @barelyhuman to have a look.

@nosferatu500 If you can create a failing test case, that would be helpful.

**Comment by barelyhuman** [comment]
I don't think a logical expression ( `x || x` or `x && x` should even trigger a check in the plugin) though I'll still go through this

**Comment by dai-shi** [comment]
Hmm, I wonder how eslint-plugin-react-hooks detect if a function is a React component.

```js
const Component = () => {
  return null;
};
```
can be a valid function component.

**Comment by nosferatu500** [comment]
Because this is valid case start from React 18, I think... 🤔

**Comment by barelyhuman** [comment]
If you can create a minimal reproduction repo for the same, that'd help with identifying this faster, @nosferatu500

**Comment by nosferatu500** [comment]
I will. But I'm not sure if I have time to do it today. Maybe tomorrow.

**Comment by barelyhuman** [comment]
> I will. But I'm not sure if I have time to do it today. Maybe tomorrow. 

Whatever works 👍🏼

**Comment by dai-shi** [comment]
> I wonder how eslint-plugin-react-hooks detect if a function is a React component.

Looks much simpler than expected.

https://github.com/facebook/react/blob/9fcaf88d58cfd942e2fdd303ae8291dbf4828969/packages/eslint-plugin-react-hooks/src/RulesOfHooks.js#L92-L106

**Comment by barelyhuman** [comment]
Yep, 
we have none of that though, since if it's a variable aligned to `useSnapshot`, we just assume it's inside a function, there's no code to solidify if it's inside one and then we have more code to check if it's inside a nested function as well, which the above code (facebook/react) doesn't really solve for us

**Comment by nosferatu500** [comment]
@barelyhuman Please take a look
[test repo](https://github.com/nosferatu500/eslint-valtio-test)

`yarn` and `yarn lint`

**Comment by barelyhuman** [comment]
Cool, that works. 

It's breaking on the de-structuring, but then every test on the repo should fail..
thanks for reporting

**Comment by barelyhuman** [comment]
so, 
weird bug, you remove `export` and it'll work and since none of the test cases had `export` we've never seen this error
